src-plugins/reformat/superResolutionToolBox.cpp

```diff
@@ -241,98 +199,58 @@ void superResolutionToolBox::setWorkspace(medAbstractWorkspace * workspace)
 void superResolutionToolBox::runProcessRequest()
 {
-    if ( d->inputs.count() != 4 ) return;
-
-    if ( !d->inputs[0] || !d->inputs[1] || !d->inputs[2] || !d->inputs[3])
-        return;
-
-    // Wait Cursor
-    this->setCursor(Qt::WaitCursor);
-    QApplication::processEvents();
-
-    d->process = new superResolutionProcess;
-    d->process->setInput(d->inputs[0],0);
-    d->process->setInput(d->inputs[1],1);
-    d->process->setInput(d->inputs[2],2);
-    d->process->setInput(d->inputs[3],3);
-
-    medRunnableProcess *runProcess = new medRunnableProcess;
-    runProcess->setProcess (d->process);
-    d->progression_stack->addJobItem(runProcess, "Progress:");
-    d->progression_stack->disableCancel(runProcess);
-
-    connect (runProcess, SIGNAL (success   (QObject*)),    this, SIGNAL (success ()));
-    connect (runProcess, SIGNAL (success   (QObject*)),    this, SLOT   (setOutput()));          //after success of process, get output
-    connect (runProcess, SIGNAL (failure   (QObject*)),    this, SIGNAL (failure ()));
-    connect (runProcess, SIGNAL (cancelled (QObject*)),    this, SIGNAL (failure ()));
-    connect (runProcess, SIGNAL (activate(QObject*,bool)), d->progression_stack, SLOT(setActive(QObject*,bool)));
-
-    medJobManager::instance()->registerJobItem(runProcess);
-    QThreadPool::globalInstance()->start(dynamic_cast<QRunnable*>(runProcess));
-
-    // Arrow Cursor
-    this->setCursor(Qt::ArrowCursor);
+    if ( d->inputVol0 && d->inputVol1 && d->inputMask0 && d->inputMask1)
+    {
+        // Wait Cursor
+        this->setCursor(Qt::WaitCursor);
+        QApplication::processEvents();
+
+        d->process = dtkAbstractProcessFactory::instance()->createSmartPointer("superResolutionProcess");
+        d->process->setInput(d->inputVol0, 0);
+        d->process->setInput(d->inputVol1, 1);
+        d->process->setInput(d->inputMask0,2);
+        d->process->setInput(d->inputMask1,3);
+
+        medRunnableProcess *runProcess = new medRunnableProcess;
+        runProcess->setProcess (d->process);
+        d->progression_stack->addJobItem(runProcess, "Progress:");
+        d->progression_stack->disableCancel(runProcess);
+
+        connect (runProcess, SIGNAL (success   (QObject*)),    this, SIGNAL (success ()));
+        connect (runProcess, SIGNAL (success   (QObject*)),    this, SLOT   (setOutput()));          //after success of process, get output
+        connect (runProcess, SIGNAL (failure   (QObject*)),    this, SIGNAL (failure ()));
+        connect (runProcess, SIGNAL (cancelled (QObject*)),    this, SIGNAL (failure ()));
+        connect (runProcess, SIGNAL (activate(QObject*,bool)), d->progression_stack, SLOT(setActive(QObject*,bool)));
+
+        medJobManager::instance()->registerJobItem(runProcess);
+        QThreadPool::globalInstance()->start(dynamic_cast<QRunnable*>(runProcess));
+
+        // Arrow Cursor
+        this->setCursor(Qt::ArrowCursor);
```

Comment:
(yeah I know this was there before, but it's been indented so it appears green on github and I review every green line :)